pkg/servicemesh/federation/exports/controller.go

```diff
@@ -70,6 +70,14 @@ func (c *Controller) RunInformer(stopChan <-chan struct{}) {
 	// no-op, informer is started by the shared factory in Federation.Start()
 }
 
+func (c *Controller) Start(stopChan <-chan struct{}) {
+	c.Controller.StartController(stopChan, c.HasSynced)
+}
+
+func (c *Controller) HasSynced() bool {
+	return c.Controller.HasSynced() && c.rm.HasSynced()
+}
+
```

Comment:
Can we get rid of the need to override these functions in every controller? Maybe by passing the `HasSynced()` function in the kube controller? Or, if it makes sense semantically, simply move the `resourceManager` to kube controller?

I'm thinking the current design is both too complex (we call StartController to which we pass our own implementation of HasSynced) and fragile (someone may implement a new controller and forget to override these method).

Reply:
Very good point. I removed these functions from discovery, imports and exports controller. Now only `ResourceManager.HasSynced()` is used in the base controller.